From 7a51d5656f8c209279e9013ab168404c71fce1e5 Mon Sep 17 00:00:00 2001
From: Alex Tu <alex.tu@canonical.com>
Date: Mon, 13 May 2019 14:53:16 +0800
Subject: [PATCH] fix: OBSOLETE_BY in DKMS.CONF (Closes #81)

cherry-pick from alextu/closes81-OBSOLETE_BY (4dbf35e)
 Conflicts:
	dkms
---
 dkms | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

Index: dkms-2.6.1/dkms
===================================================================
--- dkms-2.6.1.orig/dkms
+++ dkms-2.6.1/dkms
@@ -774,9 +774,7 @@ check_version_sanity()
     echo $"Running module version sanity check."
     local i=0
     local -a kernels_info dkms_info
-    set_module_suffix
-    read -a kernels_module < <(find_module "$lib_tree" "${4}")
-    if [ -z $kernels_module ]; then
+    if [ -n $3 ]; then
         # Magic split into array syntax saves trivial awk and cut calls.
         local -a obs=(${3//-/ })
         local -a my=(${1//-/ })
@@ -807,10 +805,11 @@ check_version_sanity()
             echo $"for future kernels above $3." >&2
             echo $"You may override by specifying --force." >&2
             return 1
-  else
-          return 0
         fi
     fi
+    set_module_suffix
+    read -a kernels_module < <(find_module "$lib_tree" "${4}")
+    [ -z $kernels_module ] || return 0
 
     if [[ ${kernels_module[1]} ]]; then
         warn $"Warning! Cannot do version sanity checking because multiple ${4}$module_suffix" \
